### PR TITLE
Update shell2session

### PR DESCRIPTION
**Pull request by bwatters-r7**
This is a work-in-progress.  The reason I've pushed it early is because it is much easier to point the testing framework at a PR than a branch.  The intended goal is to fix up this module to support migration in many more cases and verify the behavior of the override attribute.

It is out early so I can use it to test some expected changes to mettle payloads, and this module allows us to perform automated testing on meterpreter sessions for devices that have ssh access.

- [ ] Start `msfconsole`
- [ ] Get a shell (example: `use auxiliary/scanner/ssh/ssh_login` and set username, password, and RHOSTS)
- [ ] `use post/multi/manage/shell_to_meterpreter`
- [ ] `set session <session>`
- [ ] `set LHOST <IP>`
- [ ] `run`
- [ ] `sysinfo`
- [ ] **Verify** you get a valid meterpreter session
- [ ] **Verify** you get the expected architecture on your meterpreter session.  THe defualt is python if you are on linux.
- [ ] Repeat on Windows x86 Targets
- [ ] Repeat on Windows x64 Targets
- [ ] Repeat on  Linux x86 Targets
- [ ] Repeat on  Linux ARM Targets
- [ ] Repeat on  Linux MIPS Targets

As the goal is to support automated testing, I will supply the test results for this covering the test cases required.

### CODE REVIEW

**Comment by bwatters-r7** [comment]
That went poorly....

**Comment by wvu-r7** [comment]
WOAH.

File: modules/post/multi/manage/shell_to_meterpreter.rb [hunk 1]
```diff
@@ -96,13 +97,25 @@ def run
...
-        larch = [ARCH_X86]
-        vprint_status("Platform: Linux")
+        if target_info =~ /86/
+          payload_name = 'linux/x86/meterpreter/reverse_tcp'
+          larch = [ARCH_X86]
+        elsif target_info =~ /64/
+          payload_name = 'linux/x64/meterpreter/reverse_tcp'
+          larch = [ARCH_X64]
+        elsif target_info =~ /arm/
+          payload_name = 'linux/armle/meterpreter/reverse_tcp'
```
Comment (timwr) [inline modules/post/multi/manage/shell_to_meterpreter.rb]: Can you add AARCH64?
```
        elsif target_info =~ /aarch64/
          payload_name = 'linux/aarch64/meterpreter/reverse_tcp'
          larch = [ARCH_AARCH64]
```

**Comment by timwr** [comment]
Good change, I should really fix: https://github.com/rapid7/metasploit-framework/issues/8436
I think it's as simple as removing the "session should be meterpreter" check. 
Also perhaps you can rebase this change so only the relevant diff is visible?

File: modules/post/multi/manage/shell_to_meterpreter.rb [hunk 2]
```diff
@@ -96,13 +97,25 @@ def run
...
-        # Handle linux shells that were identified as 'unix'
+      vprint_status("target_info= #{target_info}")
+      vprint_status("Platform: Linux")
+      if target_info =~ /linux/i
         platform = 'linux'
-        payload_name = 'linux/x86/meterpreter/reverse_tcp'
         lplat = [Msf::Platform::Linux]
-        larch = [ARCH_X86]
-        vprint_status("Platform: Linux")
+        if target_info =~ /86/
```
Comment (timwr) [inline modules/post/multi/manage/shell_to_meterpreter.rb]: You should flip this and the /64/ check, or `Linux x86_64` will match /86/ before reaching /64/.

**Review by timwr — CHANGES_REQUESTED** [review]

File: modules/post/multi/manage/shell_to_meterpreter.rb (hunk 2 above)
Comment (timwr) [reply modules/post/multi/manage/shell_to_meterpreter.rb]: Or put x86 in the else section...

**Comment by bwatters-r7** [comment]
After some digging, I think this PR is the wrong way to go about getting the desired functionality.  I'm closing this and instead opening https://github.com/rapid7/metasploit-framework/pull/9267

**Comment by sempervictus** [comment]
If the functionality is common for both the post and exploit use cases, really local exploit and remote, it probably warrants moving up a namespace and providing accessors to both consumers via common api. All for the work BTW.

**Pull request closed without merging**